RetailCoder.VBE/Inspections/QuickFixes/PassParameterByReferenceQuickFix.cs

```diff
@@ -12,31 +14,104 @@ namespace Rubberduck.Inspections.QuickFixes
         {
-            var parameter = Context.GetText();
+            var argCtxt = GetArgContextForIdentifier(Context, _target.IdentifierName);
 
-            var parts = parameter.Split(new char[]{' '},2);
-            if (1 != parts.GetUpperBound(0))
-            {
-                return;
-            }
-            parts[0] = parts[0].Replace(Tokens.ByVal, Tokens.ByRef);
-            var newContent = parts[0] + " " + parts[1];
+            var terminalNodeImpl = GetByValNodeForArgCtx(argCtxt);
 
-            var selection = Selection.Selection;
+            var replacementLine = GenerateByRefReplacementLine(terminalNodeImpl);
 
-            var module = Selection.QualifiedName.Component.CodeModule;
+            ReplaceModuleLine(terminalNodeImpl.Symbol.Line, replacementLine);
+
+        }
+        private ArgContext GetArgContextForIdentifier(ParserRuleContext context, string identifier)
+        {
+            var procStmtCtx = (ParserRuleContext)context.Parent.Parent;
+            var procStmtCtxChildren = procStmtCtx.children;
+            for (int idx = 0; idx < procStmtCtxChildren.Count; idx++)
+            {
+                if (procStmtCtxChildren[idx] is ArgListContext)
+                {
+                    var argListContext = (ArgListContext)procStmtCtxChildren[idx];
+                    var arg = argListContext.children;
+                    for (int idxArgListCtx = 0; idxArgListCtx < arg.Count; idxArgListCtx++)
+                    {
+                        if (arg[idxArgListCtx] is ArgContext)
+                        {
+                            var name = GetIdentifierNameFor((ArgContext)arg[idxArgListCtx]);
+                            if (name.Equals(identifier))
+                            {
+                                return (ArgContext)arg[idxArgListCtx];
+                            }
+                        }
+                    }
+                }
+            }
+            return null;
+        }
+        private string GetIdentifierNameFor(ArgContext argCtxt)
+        {
+            var argCtxtChild = argCtxt.children;
+            var idRef = GetUnRestrictedIdentifierCtx(argCtxt);
+            return idRef.GetText();
+        }
+        private UnrestrictedIdentifierContext GetUnRestrictedIdentifierCtx(ArgContext argCtxt)
+        {
+            var argCtxtChild = argCtxt.children;
+            for (int idx = 0; idx < argCtxtChild.Count; idx++)
             {
-                var lines = module.GetLines(selection.StartLine, selection.LineCount);
-                var result = lines.Replace(parameter, newContent);
-                module.ReplaceLine(selection.StartLine, result);
+                if (argCtxtChild[idx] is UnrestrictedIdentifierContext)
+                {
+                    return (UnrestrictedIdentifierContext)argCtxtChild[idx];
+                }
             }
+            return null;
+        }
+        private TerminalNodeImpl GetByValNodeForArgCtx(ArgContext argCtxt)
```

Comment:
This shouldn't be needed either: an `ArgContext` has a `BYVAL` method that returns the BYVAL token if it's present, or `null` if it isn't.